### PR TITLE
Run test harness executables as the Test target

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Review by MichaelSimons — APPROVED** [review]

**Comment by mmitche** [comment]
I fixed up an issue where the exit code was not getting properly propagated, and also a local test issue where dotnet root paths ending in '\' were mucking up the command line for the harness (it removes the end quote). I tried a number of options, including MSBuild::Escape, etc. etc. It doesn't seem to make a difference at all to anything but the last character.  Go figure. Best option seems to be to trim the end \ if one exists.

**Comment by mmitche** [comment]
@MichaelSimons I believe i need another review here.

File: src/Directory.Build.targets [hunk 1]
```diff
@@ -2,4 +2,13 @@
 
   <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
 
+  <!-- Override the test target to run the test harness as a separate executable against dotnet SDK used to run these tests. -->
+  <Target Name="Test">
+    <PropertyGroup>
+      <!-- Dotnet root typically ends in a \. Trim this to avoid issues with Exec -->
```
Comment (premun) [inline src/Directory.Build.targets:8]: Is this on Windows only? Would there be a `/` on Unix?

File: src/Microsoft.DotNet.ScenarioTests.Common/Program.cs [hunk 2]
```diff
@@ -92,7 +92,7 @@ public static async Task<int> Main(string[] args)
 
             rootCommand.SetAction((ParseResult parseResult) =>
             {
-                Invoke(parseResult.GetValue(dotnetRootOption)!,
+                return Invoke(parseResult.GetValue(dotnetRootOption)!,
```
Comment (premun) [inline src/Microsoft.DotNet.ScenarioTests.Common/Program.cs:95]: Should these `!` be `?? throw new ArgumentException(...)`?

Comment (mmitche) [reply src/Microsoft.DotNet.ScenarioTests.Common/Program.cs:95]: Technically, I don't think so. The parameters are required or have defaults. But it couldn't hurt.

Comment (premun) [reply src/Microsoft.DotNet.ScenarioTests.Common/Program.cs]: QQ: is there a better help available? E.g. `Expected {dotnetRootOption.Prototype} to be set`? I know that in `MonoOptions` you can get the name and tell the user what's the real option one needs

Comment (mmitche) [reply src/Microsoft.DotNet.ScenarioTests.Common/Program.cs]: It won't even make far enough to throw ArgumentException though. You may as well throw InvalidOperationException. If you don't pass it, the parser will fail out before it even hit this point, and will provide appropriate help.

Comment (mmitche) [reply src/Microsoft.DotNet.ScenarioTests.Common/Program.cs]: The issue is that the API always returns a nullable string even if the value is guaranteed to be non-null and a valid directory (see above where the validator ensures that the file "dotnet.exe/dotnet" exists in the directory passed as dotnet-root.

**Pull request merged**